recipes/libsigcpp/2.x.x/conanfile.py

```diff
@@ -0,0 +1,79 @@
+import os
+
+from conans import ConanFile, Meson, tools
+from conans.errors import ConanInvalidConfiguration
+
+
+class LibSigCppConan(ConanFile):
+    name = "libsigcpp"
+    homepage = "https://github.com/libsigcplusplus/libsigcplusplus"
+    url = "https://github.com/conan-io/conan-center-index"
+    license = "LGPL-3.0"
+    description = (
+        "libsigc++ implements a typesafe callback system for standard C++."
+    )
+    topics = ("conan", "libsigcpp", "callback")
+    settings = "os", "compiler", "arch", "build_type"
+    options = {
+        "shared": [True, False],
+        "fPIC": [True, False],
+    }
+    default_options = {"shared": False, "fPIC": True}
+
+    _meson = None
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    @property
+    def _build_subfolder(self):
+        return "build_subfolder"
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def configure(self):
+        if self.settings.compiler == "Visual Studio":
+            raise ConanInvalidConfiguration(
+                "Visual Studio is not (yet) supported."
+            )
```

Comment:
Looking at errors, static Visual Studio is broken because __declspec(dllimport) is added at consume time (perhaps a tiny patch could fix that), but it might work with shared.

Reply:
That should be ` __declspec(dllexport)` when compiling the library itself. This is done by defining the `_WINDLL` macro. As far as I can tell from reading the build system, it is already set:

```
# Make sure we are exporting the symbols from the DLL    
if is_msvc    
  extra_sigc_cppflags += ['-DSIGC_BUILD', '-D_WINDLL']    
endif
```